open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/SubjectAreaImpl.java

```diff
@@ -42,39 +41,51 @@
-            SubjectAreaNode subjectAreaNode = new SubjectAreaNode(client);
-            SubjectAreaLine subjectAreaLine = new SubjectAreaLine(client);
-            SubjectAreaGraph subjectAreaGraph = new SubjectAreaGraphClient(client);
-
-            this.glossaryAPI = subjectAreaNode;
-            this.termAPI =  subjectAreaNode;
-            this.categoryAPI = subjectAreaNode;
-            this.relationshipAPI = subjectAreaLine;
-            this.graphAPI = subjectAreaGraph;
-            this.projectAPI = subjectAreaNode;
-        } catch (org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException e) {
-            String parameterName = "serverName or omasServerURL";
-            String parameterValue = "unknown";
-            if (serverName == null ||  "".equals(serverName)) {
-                parameterName = "serverName";
-                parameterValue = serverName;
-            }
-            if (omasServerURL == null || "".equals(omasServerURL)) {
-                parameterName = "omasServerURL";
-                parameterValue = omasServerURL;
-            }
+        this(initRestClient(serverName, omasServerURL, null, null));
+    }
+
+    /**
+     * Constructor used once a connector is created.
+     *
+     * @param serverName    serverName under which this request is performed, this is used in multi tenanting to identify the tenant
+     * @param omasServerURL - unique id for the connector instance
+     * @param userId - unique identifier for user
+     * @param password - password for user
+     * @throws InvalidParameterException one of the parameters (serverName, omasServerURL) is null or invalid.
+     */
+    public SubjectAreaImpl(String serverName, String omasServerURL, String userId, String password) throws InvalidParameterException {
+        this(initRestClient(serverName, omasServerURL, userId, password));
+    }
+
+    private SubjectAreaImpl(SubjectAreaRestClient client) {
+        SubjectAreaNode subjectAreaNode = new SubjectAreaNode(client);
+        SubjectAreaLine subjectAreaLine = new SubjectAreaLine(client);
+        SubjectAreaGraph subjectAreaGraph = new SubjectAreaGraphClient(client);
+
+        this.glossaryAPI = subjectAreaNode;
+        this.termAPI =  subjectAreaNode;
+        this.categoryAPI = subjectAreaNode;
+        this.relationshipAPI = subjectAreaLine;
+        this.graphAPI = subjectAreaGraph;
+        this.projectAPI = subjectAreaNode;
+        this.serverName = client.getServerName();
+        this.omasServerUrl = client.getServerPlatformURLRoot();
+    }
+
+    private static SubjectAreaRestClient initRestClient(String serverName,
+                                                        String omasServerURL,
+                                                        String userId,
+                                                        String password) throws InvalidParameterException {
```

Comment:
I am uncomfortable with this change . It seems to me that userid and password is basic authentication, I think we need a more considered view across the different forms to authentication.

Reply:
@davidradl  I wrote this based on super class:
<img width="1023" alt="Screenshot 2020-08-03 at 22 25 18" src="https://user-images.githubusercontent.com/35165540/89219440-745b1580-d5d8-11ea-8c9d-a1860c1df0a9.png">

Reply:
If you have any suggestions. Please suggest)

Reply:
@planetf1 Please could you look at this change, if you think it is ok - then I will merge.

Reply:
I've not looked at our use of http auth at all -- we've recently done some work on TLS security, but that is of course transport level, whilst the http authentication is another layer above. We do have a user id in the requests we pass for authorization, but that's another level up again ... I'd be inclined to put on hold without taking a broader view of how we handle http auth across all APIs. Unfortunately I don't have the bandwidth to take a look at this in the very near term (weeks)

Reply:
Just to clarify - this isn't mean to say it was wrong to propose this -- as pointed out, the superclass refers to use of the headers, so following that pattern it's very reasonable. It's just I think the truth that we put it in place initially, didn't persue, and know we need to go back to it at some point ...

Reply:
@planetf1 I think you are saying this is ok to merge but we will need to review as this area changes ? Can you confirm?

Reply:
I meant to persue http auth - and to pick up from the code that is already in place but not exploited. I don't feel the change proposed here makes sense by itself - it's only subject area, and we really need more of that architectural level review to consider how we want to support http auth in general? Plus we would want tests and docs alongside -- similar to what we did adding the TLS support (and haven't completed yet)

Reply:
@planetf1 thanks for the confirmation. @immmus please could you close out this pr and associated issue (you could leave the issue open if you want to change the words to be more of a review/investigation of the whole area).